dbms/src/Common/OptimizedRegularExpression.inl.h

```diff
@@ -725,18 +737,80 @@ void OptimizedRegularExpressionImpl<thread_safe>::replace(
     Int64 occur)
 {
     Int64 utf8_total_len = DB::UTF8::countCodePoints(reinterpret_cast<const UInt8 *>(subject), subject_size);
-    ;
+
     FunctionsRegexp::checkArgsReplace(utf8_total_len, subject_size, pos);
     FunctionsRegexp::makeReplaceOccurValid(occur);
 
+    auto num_captures = std::min(re2->NumberOfCapturingGroups() + 1, max_captures);
+    Instructions instructions = getInstructions(repl, num_captures);
```

Comment:
Looks like even if `pattern` column is a constant string ,`getInstructions` will be called for every row?

Reply:
> Looks like even if `pattern` column is a constant string ,`getInstructions` will be called for every row?

Yes, and I will memorize them in appropriate place.